### PR TITLE
add wrapper generation for packages

### PR DESCRIPTION
**Pull request by clauswilke**
Add a function `register_extendr()` that retrieves the wrapper code for a package that uses extendr. The wrappers are written to `R/extendr-wrappers.R`.

### CODE REVIEW

File: R/register_extendr.R [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      glue::glue("Package {pkg_name} cannot be loaded. No wrapper functions were generated."),
+      call. = FALSE
+    )
+  }
+}
+
+make_wrappers <- function(module_name, package_name, outfile,
+                          use_symbols = FALSE, quiet = FALSE) {
+  wrapper_function <- glue::glue("wrap__make_{module_name}_wrappers")
+  wrapper_call <- glue::glue(
```
Comment (yutannihilation) [inline R/register_extendr.R:41]: Can't we simply call .Call() here?

Comment (clauswilke) [reply R/register_extendr.R:41]: No, if we try that then R CMD check complains about a non-existent symbol.

File: R/register_extendr.R [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#' @export
+register_extendr <- function(path = ".", quiet = FALSE) {
+  x <- desc::desc(rprojroot::find_package_root_file("DESCRIPTION", path = path))
+  pkg_name <- x$get("Package")
+
+  if (!isTRUE(quiet)) {
+    message(glue::glue("Generating extendr wrapper functions for package: {pkg_name}"))
+  }
+
+  outfile = rprojroot::find_package_root_file("R", "extendr-wrappers.R", path = ".")
```
Comment (Ilia-Kosenkov) [inline R/register_extendr.R:26]: Should it be `<-` instead of `=`?

File: R/register_extendr.R (hunk 1 above)
Comment (yutannihilation) [reply R/register_extendr.R:41]: Sorry, I couldn't reproduce the complain. There's no NSE involved here so I believe there's no non-existent symbols. Am I missing something...?

commit: https://github.com/yutannihilation/rextendr/pull/1/commits/bd5b63e0565a1431c32ccd7a281a8273be4ee65f
result: https://github.com/yutannihilation/rextendr/pull/1/checks

File: R/register_extendr.R (hunk 2 above)
Comment (clauswilke) [reply R/register_extendr.R:26]: Yes, thanks for catching.

File: R/register_extendr.R (hunk 1 above)
Comment (clauswilke) [reply R/register_extendr.R]: Strange. I am certain I tried this without `eval()` first, but your code seems to work. I don't remember what I wrote that didn't work.

Comment (yutannihilation) [reply R/register_extendr.R]: Maybe you used a symbol at that time? Anyway good that it works :)

**Review by yutannihilation — APPROVED** [review]
Looks good.

**Pull request merged**